lib/iris/_data_manager.py

```diff
@@ -365,6 +386,14 @@ def data(self, data):
         # Always reset the realised dtype, as the managed data has changed.
         self._realised_dtype = None
 
+        # Reset the fill-value appropriately.
+        if ma.isMaskedArray(data):
+            # Align the data manager fill-value with the numpy fill-value.
```

Comment:
@pp-mo I disagree.

I think there is real value is being able to do `cube.data = mask_array` to set **both** the data to a real masked array and set the `cube.fill_value` to be the masked array `fill_value` - note that the numpy default `fill_value` is handled appropriately, see #2548.

The behaviour supported in this PR is opposed to the following:
```python
cube.data = masked_array
cube.fill_value = masked_array.fill_value
```
or
```python
cube.replace(masked_data, fill_value=masked_data.fill_value)
```
Actually, with `replace`, regardless, you need to always provide a `fill_value`, otherwise it will clear the cube `fill_value`.

Also, with #2548 (correcting an oversight that existed before this PR port), it's possible to get the same `fill_value` setting behaviour from a masked array via the `DataManager` constructor e.g.
```python
cube = Cube(masked_array)
```
will set the `cube.fill_value` to be the masked array `fill_value`, otherwise you'd have to do:
```python
cube = Cube(masked_array, fill_value=masked_array.fill_value)
```